homeassistant/components/tessie/update.py

```diff
@@ -0,0 +1,51 @@
+"""Update platform for Tessie integration."""
+from __future__ import annotations
+
+from homeassistant.components.update import UpdateEntity, UpdateEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import TessieDataUpdateCoordinator
+from .entity import TessieEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie Update platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(TessieUpdateEntity(coordinator) for coordinator in coordinators)
+
+
+class TessieUpdateEntity(TessieEntity, UpdateEntity):
+    """Tessie Updates entity."""
+
+    _attr_supported_features = (
+        UpdateEntityFeature.INSTALL | UpdateEntityFeature.PROGRESS
+    )
+
+    def __init__(
+        self,
+        coordinator: TessieDataUpdateCoordinator,
+    ) -> None:
+        """Initialize the Update."""
+        super().__init__(coordinator, "update")
+
+    @property
+    def installed_version(self) -> str:
+        """Return the current app version."""
+        # Discard build from version number
+        return self.get("vehicle_state_car_version").split(" ")[0]
```

Comment:
This could be problematic when `.get` returns `None`.

Reply:
Yeah, in theory it never will, because this field always exists, but I did wonder if I should None check it, or not use get.